presto-main/src/main/java/io/prestosql/sql/planner/iterative/rule/PushLimitThroughProject.java

```diff
@@ -32,6 +32,7 @@
     private static final Capture<ProjectNode> CHILD = newCapture();
 
     private static final Pattern<LimitNode> PATTERN = limit()
+            .matching(limit -> !limit.isWithTies())
```

Comment:
Why?

Reply:
Because I mistakenly assumed that Limit with ties would never match anyway.
I'll add it to the rule.
BTW if I was right, then what was the proper thing to do? Exclude it in the pattern? Or make the rule so that it takes account of all possibilities?

Reply:
Excluding in the pattern is fine, assuming that's what we need to do.